### ISSUE TITLE
Migrate to stardoc, fix docs/ directory

### ISSUE THREAD

**Issue by tmc** [bazelbuild/rules_python#144]
on current HEAD (f3a6a8d00a51a1f0e6d61bc7065c19fea2b3dd7a)

after `bazel clean --expunge; bash update_doc.sh`:
```sh
$ git diff --stat
 docs/index.html | 130 +---------------------------------------------------------------------------------------------------------------------------------
 docs/index.md   | 111 ---------------------------------------------------------------------------------------------------------------
 docs/main.css   |   4 +---
 3 files changed, 2 insertions(+), 243 deletions(-)
```

**Comment by brandjon** [comment]
This will be fixed by #213 shortly, but running the new docgen workflow requires a change to Bazel that will be released in 0.29. In the meantime I'll merge my fix, regenerate the docs using Bazel at head, and mark this bug fixed once 0.29 is released.

**Comment by brandjon** [comment]
Leaving open, as explained above.
